src/App.tsx

```diff
@@ -15,47 +16,125 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[],
+  sortType: SortType,
+  reversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((first, next) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return first.localeCompare(next);
+
+      case SortType.LENGTH:
+        return first.length - next.length;
+
+      case SortType.NONE:
+        return 0;
+
+      default:
+        throw new Error('Wrong sort type');
+        break;
+    }
+  });
+
+  if (reversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [reversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const sortAlphabetically = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
```

Comment:
You don't really need to make a separate function for both set functions as you can call them in onClick handler